extension/googleclientauthextension/factory_test.go

```diff
@@ -19,32 +19,43 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/extension"
 )
 
+// newFactory creates a factory for the GCP Auth extension.
+func newFactory() extension.Factory {
```

Comment:
I don't think this is necessary.  You should just call the public API (CreateDefaultConfig and CreateExtension) as needed to test them.

Reply:
Yup you're totally right, updated all suggestions above